File: 5 all of them -C#/Kattis/Kattis/I_No_Edges.cs
using Kattis.IO;
using Microsoft.Win32.SafeHandles;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.Design;
using System.Diagnostics;
using System.Linq;
using System.Numerics;
using System.Threading;

namespace Kattis
{
    class Node
    {
        public Boolean isWalkable;
        public Tuple<int, int> pos;
        public Node(bool isWalkable, Tuple<int, int> pos)
        {
            this.isWalkable = isWalkable;
            this.pos = pos;
        }
        public override string ToString()
        {
            return isWalkable ? "." : "#";
        }
        public void PrintSelf()
        {
            Console.WriteLine(this.ToString() + "  -  " +pos);
        }
    }
    class Case{
        public Node[][] nodes;
        public Node s;
        public Node t;
        public int n;

        public Case(int numberOfRowsCols)
        {
            n = numberOfRowsCols;
            this.nodes =  new Node[numberOfRowsCols][];
        }
        public void addLine(char[] line, int lineIndex)
        {
            var currNodes = new Node[line.Length];
            for(int i=0; i<line.Length; i++)
                currNodes[i] = new Node(line[i] == '.', Tuple.Create(lineIndex, i));
            nodes[lineIndex] = currNodes;
        }
        public void finish()
        {
            s = nodes[0][0];
            t = nodes[n - 1][n - 1];
        }
        public void printCase()
        {
            s.PrintSelf();
            foreach (Node[] nodeRow in nodes){
                foreach (Node node in nodeRow)
                    Console.Write(node);
                Console.WriteLine("");
            }
        }
    }

    class I_No_Edges
    {
        static void Main(string[] args)
        {
            var currCase = parse_start();
            currCase.finish();
            //currCase.printCase();

            var dijk = dijkstraFindNumberOfPaths(currCase.s, currCase.nodes, currCase.n);
            if (dijk[currCase.n - 1][currCase.n - 1] > 0) { 
                BigInteger mod = BigInteger.Pow(2, 31) - 1;
                Console.WriteLine(dijk[currCase.n - 1][currCase.n - 1] % mod);
            }
            else
            {
                var foundPath = bfs(currCase.nodes);
                if(foundPath) Console.WriteLine("THE GAME IS A LIE");
                else Console.WriteLine("INCONCEIVABLE");
            }
        }

        static Case parse_start()
        {
            Scanner s = new Scanner();
            Case currCase = new Case(s.NextInt());

            int line_number = 0;        //first is numBlocks above
            //while (s.HasNext())
            while(line_number < currCase.n)
            {
                line_number++;
                string line = s.Next();

                var split = line.ToCharArray();
                currCase.addLine(split, line_number - 1);
            }
            return currCase;
        }

        static BigInteger[][] dijkstraFindNumberOfPaths(Node s, Node[][] nodes, int n)
        {
            var visited = new BigInteger[n][];
            for (int i = 0; i < n; i++)
                visited[i] = new BigInteger[n];
            visited[0][0] = 1;
            for (int row = 0; row < n; row++)
            {
                for (int col = 0; col < n; col++)
                {
                    var curr = nodes[row][col];
                    if (!curr.isWalkable) continue;
                    BigInteger pathsToN = 0;
                    if (row > 0)        pathsToN += visited[row - 1][col];
                    else if (col == 0)  continue;
                    if (col > 0)        pathsToN += visited[row][col - 1];
                    visited[row][col] = pathsToN;
                }
            }


            return visited;
        }
        static List<Tuple<int, int>> edges(int row, int col, int n) {
            var res = new List<Tuple<int, int>>();
            if (row > 0) res.Add(Tuple.Create(row - 1, col));
            if (row < n - 1) res.Add(Tuple.Create(row + 1, col));
            if (col > 0) res.Add(Tuple.Create(row, col-1));
            if (col < n - 1) res.Add(Tuple.Create(row, col+1));
            return res;
        }

        static bool bfs(Node[][] nodes)
        {
            int n = nodes.Length;
            var visited = new bool[n][];
            for (int i = 0; i < n; i++)
                visited[i] = new bool[n];
            visited[0][0] = true;
            var frontier = new Stack<Tuple<int,int>>();
            foreach (var e in edges(0, 0, n)) 
                frontier.Push(e);
            while (frontier.Count > 0)
            {
                var e = frontier.Pop();
                if (visited[e.Item1][e.Item2]) continue;
                if (!nodes[e.Item1][e.Item2].isWalkable) continue;
                if (e.Item1 == n - 1 && e.Item2 == n - 1)
                    return true;

                visited[e.Item1][e.Item2] = true;
                foreach (var eNew in edges(e.Item1, e.Item2, n))
                    frontier.Push(eNew);
                
            }

            return false;
        }

    }
}

File: 5 all of them -python/I_No_Edges.py
from sys import stdin
import math
from functools import lru_cache
import copy
from heapq import *
from queue import LifoQueue
#Single source shortest path, negative weights
#Bellan ford algorithm

class Node(object):
    def __init__(self, value, pos):
        self.value = value
        self.pos = pos
    
    def __str__(self):
        return str(self.value) + str(self.pos)
        
    def printSelf(self):
        print(self.value)

class Case(object):
    def __init__(self, numberOfNodes):
        self.nodes      = [None for i in range(numberOfNodes)]     #ignore nodes[0]
        self.s          = None
    
    def addLine(self, line, lineIndex):
        lst = [None for i in range(len(line))]
        for i in range(len(line)):
            lst[i] = Node(line[i], (lineIndex, i))
        self.nodes[lineIndex] = lst
    
    def finish(self):
        n = len(self.nodes[0])
        self.s = self.nodes[0][0]
        self.t = self.nodes[n-1][n-1]
    
    def printCase(self):
        self.s.printSelf()
        for l in self.nodes:
            print(*[n.value for n in l])     
        
                

def parse_start():
    case = None
    for (line_number, line) in enumerate(stdin):
        split = line.rstrip('\r\n').split(" ")
        if line_number == 0:
            case = Case(int(split[0]))
            continue
        split = list(split[0])
        case.addLine(split, line_number-1)
    case.finish()
    return case

def RemoveIfThere(set, item):
    if item in set:
        set.discard(item)
 

def dijkstraFindNumberOfPaths(s, nodes):
    n = len(nodes)
    visited = [[0 for _ in range(n)] for _ in range(n)]
    frontier = []
    visited[0][0] = 1
    frontierNotTaken = []
    for row in range(n):
        for col in range(n):
            curr = nodes[row][col]
            if curr.value == "#":
                continue
            
            pathsToN = 0
            if row>0:
                pathsToN += visited[row-1][col]
            elif col == 0:
                continue        #row=0, col=0   //don't override s
            if col>0:
                pathsToN += visited[row][col-1]
            visited[row][col] = pathsToN
            if pathsToN == 0:
                continue
            for e in edgesAboveAndRight(row,col,n):
                frontierNotTaken.append(e)        # pick closets to end first
    return visited, frontierNotTaken

def edgesAboveAndRight(row,col,n):
    res = []
    if row>0: 
        res.append((row-1,col)) 
    if col>0: 
        res.append((row,col-1))
    return res

def edges(row,col,n):
    res = edgesAboveAndRight(row,col,n)
    if row<n-1: 
        res.append((row+1,col)) 
    if col<n-1: 
        res.append((row,col+1))
    return res

def bfs(nodes, visited, frontier):
    n = len(nodes)  
    while (frontier != []):
        e = frontier.pop()
        (row,col) = e
        if visited[row][col] > 0:
            continue
        if nodes[row][col].value == "#":
            continue
        if  row == n-1 and col == n-1:
            return True
        visited[row][col] = 1 
        for e in edges(row,col,n):
            frontierNotTaken.append(e)
    return False

def printPath(path):
    print("Path")
    for k in path:
        (t,e) = path[k]
        if e is None: print("   "+str(t)+" : None")
        else:         print("   "+str(t)+" : "+e.toString())



case = parse_start()
n = len(case.nodes)
#import time
#start_time = time.time()
dijk, frontierNotTaken = dijkstraFindNumberOfPaths(case.s, case.nodes)
#print("--- %s seconds ---" % (time.time() - start_time))
if dijk[n-1][n-1]>0:
    print(dijk[n-1][n-1] % ((2**31) -1))    
else:
    path = bfs(case.nodes, dijk, frontierNotTaken)
    if path:
        print("THE GAME IS A LIE")
    else:
        print("INCONCEIVABLE")
